modules/dmrpp_module/DmrppArray.cc

```diff
@@ -1575,13 +1575,95 @@ void DmrppArray::read_chunks_with_linked_blocks() {
         this->insert_chunk_unconstrained(chunk, 0, 0, array_shape, 0,chunk_shape,chunk_origin);
 
     }
+    set_read_p(true);
 
 }
 
 void DmrppArray::read_chunks_with_linked_blocks_constrained() {
 
-    // Will handle this in a separate ticket.
+    reserve_value_capacity_ll(get_size(true));
+    char *dest_buf = this->get_buf();
```

Comment:
I was going to say that you don't need to use 'this->' (which I know you know) but I think it's good here because this is complicated and helps make it clear that the buffer and shape are for the chunk.